Content.Client/_DV/CosmicCult/UI/Monument/MonumentMenu.xaml.cs

```diff
@@ -0,0 +1,190 @@
+        UpdateEntropy(state);
+        UpdateGlyphs();
+        UpdateInfluences(state);
+    }
+
+    // Update all the entropy fields
+    private void UpdateBar(MonumentBuiState state)
+    {
+        var percentComplete = ((float) state.CurrentProgress / (float) state.TargetProgress) * 100f; //too many parenthesis & probably unnecessary float casts but I'm not taking any chances
+
+        percentComplete = Math.Min(percentComplete, 100);
+
+        CultProgressBar.Value = percentComplete;
+
+        ProgressBarPercentage.Text = Loc.GetString("monument-interface-progress-bar", ("percentage", percentComplete.ToString("0")));
+    }
+
+    // Update all the entropy fields
+    private void UpdateEntropy(MonumentBuiState state)
+    {
+        var availableEntropy = "thinking emoji"; //if you see this, problem.
+        if (_ent.TryGetComponent<CosmicCultComponent>(_player.LocalEntity, out var cultComp))
+        {
+            availableEntropy = cultComp.EntropyBudget.ToString();
+        }
+
+        var entropyToNextStage = Math.Max(state.TargetProgress - state.CurrentProgress, 0);
+        var min = entropyToNextStage == 0 ? 0 : 1; //I have no idea what to call this. makes it so that it shows 0 crew for the final stage but at least one at all other times
+        var crewToNextStage = (int)Math.Max(Math.Round((double)entropyToNextStage / _ccEntropyValue, MidpointRounding.ToPositiveInfinity), min); //force it to be at least one
+
+        AvailableEntropy.Text = Loc.GetString("monument-interface-entropy-value", ("infused", availableEntropy));
+        EntropyUntilNextStage.Text = Loc.GetString("monument-interface-entropy-value", ("infused", entropyToNextStage.ToString()));
+        CrewToConvertUntilNextStage.Text = crewToNextStage.ToString();
+    }
+
+    // Update all the glyph buttons
+    private void UpdateGlyphs()
+    {
+        GlyphContainer.RemoveAllChildren();
+        foreach (var glyph in _glyphPrototypes)
+        {
+            var boxContainer = new BoxContainer();
+            var unlocked = _unlockedGlyphProtoIds.Contains(glyph.ID);
+            var button = new Button
+            {
+                HorizontalExpand = true,
+                StyleClasses = { StyleBase.ButtonSquare },
+                ToolTip = Loc.GetString(glyph.Tooltip),
+                Group = _glyphButtonGroup,
+                Pressed = glyph.ID == _selectedGlyphProtoId,
+                Disabled = !unlocked,
+                Modulate = !unlocked ? Color.Gray : Color.White,
+            };
+            button.OnPressed += _ => _selectedGlyphProtoId = glyph.ID;
+            var glyphIcon = new TextureRect
+            {
+                Texture = _sprite.Frame0(glyph.Icon),
+                TextureScale = new Vector2(2f, 2f),
+                Stretch = TextureRect.StretchMode.KeepCentered,
+            };
+            boxContainer.AddChild(button);
+            button.AddChild(glyphIcon);
+            GlyphContainer.AddChild(boxContainer);
```

Comment:
Don't think this makes a difference but it's nicer to reason about when doing these in order

```suggestion
            button.AddChild(glyphIcon);
            boxContainer.AddChild(button);
            GlyphContainer.AddChild(boxContainer);
```